status/views.py

```diff
@@ -0,0 +1,121 @@
+"""
+Status checks for LORE.
+
+Notes:
+
+* Useful messages are logged, but NO_CONFIG is returned whether
+  settings are missing or invalid, to prevent information leakage.
+* Different services provide different information, but all should return
+  "up," DOWN, or NO_CONFIG for the "status" key.
+"""
+
+import re
+from datetime import datetime
+import logging
+
+from django.conf import settings
+from django.http import JsonResponse, Http404
+from elasticsearch import Elasticsearch, ConnectionError as ESConnectionError
+from psycopg2 import connect, OperationalError
+from redis import StrictRedis, ConnectionError as RedisConnectionError
+
+log = logging.getLogger(__name__)
+
+UP = "up"
+DOWN = "down"
+NO_CONFIG = "no config found"
+HTTP_OK = 200
+SERVICE_UNAVAILABLE = 503
+
+
+def get_pg_info():
+    """Check PostgreSQL connection."""
+    try:
+        conf = settings.DATABASES['default']
+    except (AttributeError, KeyError):
+        log.error("No PostgreSQL connection info found in settings.")
+        return {"status": NO_CONFIG}
+    try:
+        start = datetime.now()
+        connection = connect(
+            database=conf["NAME"], user=conf["USER"], host=conf["HOST"],
+            port=conf["PORT"], password=conf["PASSWORD"]
+        )
+        micro = (datetime.now() - start).microseconds
+        connection.close()
+    except (OperationalError, KeyError):
+        log.error("Invalid PostgreSQL connection info in settings: %s", conf)
+        return {"status": DOWN}
+    return {"status": UP, "response_microseconds": micro}
+
+
+def get_redis_info():
+    """Check Redis connection."""
+    pat = re.compile(r'redis://([^:]+):(\d+)/(\d+)$')
+    try:
+        url = settings.BROKER_URL
+    except AttributeError:
+        log.error("No Redis connection info found in settings.")
+        return {"status": NO_CONFIG}
+    try:
+        match = pat.match(url)
+    except TypeError:
+        match = None
+    if match is None:
+        log.error("Invalid Redis connection info found in settings: %s", url)
+        return {"status": NO_CONFIG}
+    host, port, database = match.groups()
+    rdb = StrictRedis(host=host, port=port, db=database)
+    start = datetime.now()
+    try:
+        info = rdb.info()
+    except RedisConnectionError:
```

Comment:
I appear to get the same result with Redis, removing the redis link in docker-compose appears to cause the page to infinitely try.